soc/arm/atmel_sam0/common/soc_samr34.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021 Argentum Systems Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/pinmux.h>
+#include <soc.h>
+
+static int soc_pinmux_init(const struct device *dev)
+{
+	const struct device *muxb = DEVICE_DT_GET(DT_NODELABEL(pinmux_b));
+	const struct device *muxc = DEVICE_DT_GET(DT_NODELABEL(pinmux_c));
```

Comment:
Pinmux driver is in EOL. This means that it should not be encouraged to be at SoC directory in my view.
Please, keep at board for now. As soon SPI driver get pinctrl capabilities all spi pinmux will be dropped, see #39740

Reply:
> Please, keep at board for now.

Even though this is SoC specific, not board specific?

> As soon SPI driver get pinctrl capabilities all spi pinmux will be dropped

SPI supporting pinctrl doesn't fully resolve the issue here... when the internal radio is not used, it must be held in reset with `#CS` high to avoid unpredictable radio behaviour and excessive power usage - this is called out explicitly in the datasheet.

I'd be quite uncomfortable putting this code on the board layer due to this...

Are we able to get similar (raw GPIO) functionality from `pinctrl.h`? A quick look suggests no...